File: assignments/a1/q8/Java/NoComment.java
//NoComment:top//
//NoComment:import//
import java.util.*;

// <line>NoComment ::= <rule>
public class NoComment extends Line /*NoComment:class*/ {

    public static final String $className = "NoComment";
    public static final String $ruleString =
        "<line>NoComment ::= <rule>";

    public Rule rule;

    public NoComment(Rule rule) {
//NoComment:init//
        this.rule = rule;
    }

    public static NoComment parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<line>NoComment", scn$.lno);
        Rule rule = Rule.parse(scn$, trace$);
        return new NoComment(rule);
    }

//NoComment//
}

File: assignments/a1/q8/Python/NoComment.py
#NoComment:top#
#NoComment:import#

# <line>NoComment ::= <rule>
class NoComment(Line): #NoComment:class#

    className = "NoComment"
    ruleString = "<line>NoComment ::= <rule>"
    rule = None

    def __init__(rule):
        #NoComment:init#
        self.rule = rule

#NoComment#
